src/native/corehost/hostpolicy/deps_format.cpp

```diff
@@ -39,6 +39,55 @@ namespace
 
         return path;
     }
+
+    void populate_rid_fallback_graph(const json_parser_t::value_t& json, deps_json_t::rid_fallback_graph_t& rid_fallback_graph)
+    {
+        const auto& json_object = json.GetObject();
+        if (json_object.HasMember(_X("runtimes")))
+        {
+            for (const auto& rid : json[_X("runtimes")].GetObject())
+            {
+                auto& vec = rid_fallback_graph[rid.name.GetString()];
+                const auto& fallback_array = rid.value.GetArray();
+                vec.reserve(fallback_array.Size());
+                for (const auto& fallback : fallback_array)
+                {
+                    vec.push_back(fallback.GetString());
+                }
+            }
+        }
+
+        if (trace::is_enabled())
+        {
+            trace::verbose(_X("The rid fallback graph is: {"));
+            for (const auto& rid : rid_fallback_graph)
+            {
+                trace::verbose(_X("%s => ["), rid.first.c_str());
+                for (const auto& fallback : rid.second)
+                {
+                    trace::verbose(_X("%s, "), fallback.c_str());
+                }
+                trace::verbose(_X("]"));
+            }
+            trace::verbose(_X("}"));
+        }
+    }
+}
+
+deps_json_t::rid_fallback_graph_t deps_json_t::get_rid_fallback_graph(pal::string_t& deps_path)
+{
+    rid_fallback_graph_t rid_fallback_graph;
+
+    trace::verbose(_X("Getting RID fallback graph for deps file... %s"), deps_path.c_str());
+    if (!bundle::info_t::config_t::probe(deps_path) && !pal::realpath(&deps_path, true))
+        return rid_fallback_graph;
+
+    json_parser_t json;
+    if (!json.parse_file(deps_path))
+        return rid_fallback_graph;
```

Comment:
This logic is (and should be) identical to the beginning of `load` - we should share that somehow (it would also add a trace to the case where we can't find the deps file, which will be helpful for diagnostics).

Reply:
Yeah, I did take it from there. I think `json_parser_t` scoping was a bit annoying (admittedly I didn't try super hard to reconcile the two) - maybe I'll just pull out the existence check for sharing, since the actual parse is straight-forward and `parse_file` itself has logging.